metadata-ingestion/src/datahub/ingestion/source/looker/lookml_source.py

```diff
@@ -1673,7 +1595,30 @@ def _get_upstream_lineage(
                                 make_schema_field_urn(
                                     upstream_dataset_urn, upstream_field
                                 )
-                                for upstream_field in field.upstream_fields
+                                for upstream_field in cast(
+                                    List[str], field.upstream_fields
+                                )
+                            ],
+                            downstreamType=FineGrainedLineageDownstreamType.FIELD,
+                            downstreams=[
+                                make_schema_field_urn(
+                                    looker_view.id.get_urn(self.source_config),
+                                    field.name,
+                                )
+                            ],
+                        )
+                        fine_grained_lineages.append(fine_grained_lineage)
+            else:
+                # View is defined as SQL
+                for field in looker_view.fields:
+                    if field.upstream_fields:
+                        fine_grained_lineage = FineGrainedLineageClass(
+                            upstreamType=FineGrainedLineageUpstreamTypeClass.FIELD_SET,
+                            upstreams=[
+                                make_schema_field_urn(cll_ref.table, cll_ref.column)
+                                for cll_ref in cast(
+                                    List[ColumnRef], field.upstream_fields
```

Comment:
why is this cast required?

in general, anytime you're using cast, there's a decent chance you're doing something wrong or need an `if thing is None` check

Reply:
Please check ViewField.upstream_fields type